cuckoo/data/analyzer/windows/modules/auxiliary/human.py

```diff
@@ -151,14 +188,76 @@ def run(self):
         if "human.click_buttons" in self.options:
             self.do_click_buttons = int(self.options["human.click_buttons"])
 
+        if "human.change_windows" in self.options:
+            self.do_change_windows = int(self.options["human.change_windows"])
+
+        if self.do_change_windows:
+            # Pick some random procs and spawn them
+            spawn = [
+                "calc.exe", "notepad.exe", "iexplore.exe", "cmd.exe",
+                "explorer.exe", "wmplayer.exe"
+            ]
+            # Spawn 2 random procs
+            hwnd = USER32.GetForegroundWindow()
+            for i in xrange(2):
+                app = random.choice(spawn)
+                log.debug("[HUMAN.CHANGE_WINDOWS] Spawning %s", app)
+                SHELL32.ShellExecuteA(hwnd, "open", app, "", "", SW_SHOW)
+
         while self.do_run:
             if self.do_click_mouse:
                 click_mouse()
 
             if self.do_move_mouse:
                 move_mouse()
 
+            if self.do_change_windows:
+                change_windows()
+
             if self.do_click_buttons:
                 USER32.EnumWindows(EnumWindowsProc(foreach_window), 0)
 
             KERNEL32.Sleep(1000)
+
+    @staticmethod
+    def handle_32770(frameHandle):
```

Comment:
Overall this works pretty well (I'm running variant of this code on a test system). One issue we hit was this would continue to trigger right + enter even after the "View Downloads" run button in IE11 was clicked to run the file. In some cases it would cycle through and keep clicking other things in that "View Downloads" window, most of which unrelated and undesirable.

I had resolved by adding a flag for "view_downloads_clicked" that I set once doing right right + enter enter and checking before, although this may have other side affects if click needs to happen multiple times. My modifications below.

```
@staticmethod
    def handle_32770(frameHandle):
        """
          Handle IE11's "View Downloads" window
        """
        if not Human.view_downloads_clicked:
            log.debug("Setting %s as foreground window", frameHandle)
            USER32.SetForegroundWindow(frameHandle)
            # Press RIGHT + ENTER to run the file
            log.debug("Waiting 15 seconds, hopefully download finishes in time")
            KERNEL32.Sleep(15000)

            log.debug("Sending RIGHT + ENTER...")

            USER32.keybd_event(VK_RIGHT, 0x4D, KEYEVENTF_EXTENDEDKEY | 0, 0)
            USER32.keybd_event(VK_RIGHT, 0x4D, KEYEVENTF_EXTENDEDKEY | KEYEVENTF_KEYUP, 0)
            USER32.keybd_event(VK_RETURN, 0x1C, KEYEVENTF_EXTENDEDKEY | 0, 0)
            USER32.keybd_event(VK_RETURN, 0x1C, KEYEVENTF_EXTENDEDKEY | KEYEVENTF_KEYUP, 0)

            Human.view_downloads_clicked = True
```